### PR TITLE
feat(core): impl team workspace

### PR DESCRIPTION
**Pull request by JimmFly**
AF-1738 AF-1735 AF-1731 AF-1721 AF-1717 AF-1736 AF-1727 AF-1719 AF-1877
UI for team workspaces :
- add upgrade to team & successful upgrade page ( `/upgrade-to-team` & `/upgrade-success/team`)
- update team plans on pricing page ( settings —> pricing plans )
- update reaching the usage/member limit modal
- update invite member modal
- update member CRUD options

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#8920** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8920?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3  dependent PRs ([#9026](https://github.com/toeverything/AFFiNE/pull/9026) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9026?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#9043](https://github.com/toeverything/AFFiNE/pull/9043) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9043?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#9085](https://github.com/toeverything/AFFiNE/pull/9085) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9085?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8920?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/component/src/components/member-components/invite-team-modal/link-invite.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    [menuItems, selectedValue]
+  );
+
+  //TODO(@JimmFly): implement team feature
+  const onGenerate = useCallback(() => {
+    setInvitationLink('ggsimida');
+  }, []);
+
+  const onCopy = useCallback(() => {
+    navigator.clipboard
```
Comment (forehalo) [inline packages/frontend/component/src/components/member-components/invite-team-modal/link-invite.tsx]: use `copyTextToClipboard`

File: packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx [hunk 2]
```diff
@@ -92,6 +92,69 @@ export const CancelAction = ({
...
+  const authService = useService(AuthService);
+  const downgradeNotify = useDowngradeNotify();
+
+  const downgrade = useAsyncCallback(async () => {
+    try {
+      const account = authService.session.account$.value;
+      const prevRecurring = teamSubscription?.recurring;
+      setIsMutating(true);
+      await subscription.cancelSubscription(idempotencyKey);
+      subscription.revalidate();
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx]:
```suggestion
      await subscription.waitForRevalidation();
```

File: packages/frontend/core/src/modules/workspace-engine/impls/cloud.ts [hunk 3]
```diff
@@ -229,10 +234,13 @@ class CloudWorkspaceFlavourProvider implements WorkspaceFlavourProvider {
     const cloudData = await cloudStorage.pull(id);
 
     const isOwner = await this.getIsOwner(id, signal);
```
Comment (EYHN) [inline packages/frontend/core/src/modules/workspace-engine/impls/cloud.ts]: can we merge them into one request ?

File: packages/frontend/graphql/src/graphql/get-is-admin.gql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+query getIsAdmin($workspaceId: String!) {
+  isAdmin(workspaceId: $workspaceId)
+}
```
Comment (EYHN) [inline packages/frontend/graphql/src/graphql/get-is-admin.gql:3]: try this

```suggestion
query getWorkspaceInfo($workspaceId: String!) {
  isAdmin(workspaceId: $workspaceId)
  isOwner(workspaceId: $workspaceId)
  isTeam(workspaceId: $workspaceId)
}
```

File: packages/frontend/core/src/modules/cloud/entities/subscription.ts [hunk 5]
```diff
@@ -42,6 +42,11 @@ export class Subscription extends Entity {
       ? subscriptions.find(sub => sub.plan === SubscriptionPlan.AI)
       : null
   );
+  team$ = this.subscription$.map(subscriptions =>
```
Comment (EYHN) [inline packages/frontend/core/src/modules/cloud/entities/subscription.ts:45]: here this UserSubscription, you need a WorkspaceSubscription

File: packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/billing/index.tsx [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+} from '../../general-setting/plans/actions';
+import * as styles from './styles.css';
+
+export const WorkspaceSettingBilling = ({
+  workspaceMetadata,
+}: {
+  workspaceMetadata: WorkspaceMetadata;
+}) => {
+  const t = useI18n();
+  const workspace = useWorkspace(workspaceMetadata);
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/billing/index.tsx]:
```suggestion
  const workspace = useService(WorkspaceService).workspace;
```

File: packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/billing/index.tsx [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+  const workspaceInfo = useWorkspaceInfo(workspaceMetadata);
+  const subscriptionService = useService(SubscriptionService);
+  const team = useLiveData(subscriptionService.subscription.team$);
+  const title = workspaceInfo?.name || 'untitled';
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/billing/index.tsx]:
```suggestion
  const title = useLiveData(workspace.name$) || 'untitled'
```

File: packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/new-workspace-setting-detail/members/index.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import { CloudWorkspaceMembersPanel } from './cloud-members-panel';
+import * as styles from './styles.css';
+
+export const MembersPanel = ({
+  onChangeSettingState,
+}: {
+  onChangeSettingState: (settingState: SettingState) => void;
+}): ReactElement | null => {
+  const workspace = useService(WorkspaceService).workspace;
+  const isTeam = useWorkspaceInfo(workspace.meta)?.isTeam;
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/new-workspace-setting-detail/members/index.tsx:19]: maybe use team subscription, useWorkspaceInfo just for workspace list

File: packages/frontend/core/src/modules/cloud/entities/subscription.ts (hunk 5 above)
Comment (JimmFly) [reply packages/frontend/core/src/modules/cloud/entities/subscription.ts:45]: I can't seem to find the relevant backend interface, can you tell me which one it is?

Comment (JimmFly) [reply packages/frontend/core/src/modules/cloud/entities/subscription.ts:45]: Open another PR to handle this issue

File: packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/new-workspace-setting-detail/members/index.tsx (hunk 8 above)
Comment (JimmFly) [reply packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/new-workspace-setting-detail/members/index.tsx:19]: Open another PR to handle this issue

File: packages/frontend/core/src/modules/cloud/entities/subscription.ts (hunk 5 above)
Comment (forehalo) [reply packages/frontend/core/src/modules/cloud/entities/subscription.ts:45]: packages/backend/server/src/schema.gql#978, it should be fixed in this PR, the logic is wrong

File: packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx [hunk 9]
```diff
@@ -45,7 +45,7 @@ export const CancelAction = ({
       const prevRecurring = subscription.pro$.value?.recurring;
       setIsMutating(true);
       await subscription.cancelSubscription(idempotencyKey);
-      subscription.revalidate();
+      await subscription.waitForRevalidation();
       await subscription.isRevalidating$.waitFor(v => !v);
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx]: remove this line

File: packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx [hunk 10]
```diff
@@ -92,6 +92,69 @@ export const CancelAction = ({
...
+  const downgradeNotify = useDowngradeNotify();
+
+  const downgrade = useAsyncCallback(async () => {
+    try {
+      const account = authService.session.account$.value;
+      const prevRecurring = teamSubscription?.recurring;
+      setIsMutating(true);
+      await subscription.cancelSubscription(idempotencyKey);
+      await subscription.waitForRevalidation();
+      await subscription.isRevalidating$.waitFor(v => !v);
```
Comment (EYHN) [inline packages/frontend/core/src/desktop/dialogs/setting/general-setting/plans/actions.tsx]: remove this line

File: packages/frontend/core/src/modules/workspace-engine/impls/cloud.ts [hunk 11]
```diff
@@ -228,11 +230,12 @@ class CloudWorkspaceFlavourProvider implements WorkspaceFlavourProvider {
     const localData = await docStorage.doc.get(id);
     const cloudData = await cloudStorage.pull(id);
 
-    const isOwner = await this.getIsOwner(id, signal);
+    const info = await this.getWorkspaceInfo(id, signal);
 
     if (!cloudData && !localData) {
       return {
-        isOwner,
+        isOwner: info.isOwner,
```
Comment (EYHN) [inline packages/frontend/core/src/modules/workspace-engine/impls/cloud.ts:237]:
```suggestion
        isOwner: info.isOwner,
        isAdmin: info.isAdmin,
```

**Review by forehalo — APPROVED** [review]

**Pull request merged**